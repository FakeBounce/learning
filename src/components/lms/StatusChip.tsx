import { Box, Chip, IconButton } from '@mui/material';
import Iconify from '@src/components/iconify/Iconify';
import { Theme } from '@mui/material/styles';
import React, { memo, ReactNode } from 'react';

interface StatusChipProps {
  isActive?: boolean;
  activatedText?: ReactNode;
  handleClick: (event: React.MouseEvent<HTMLElement>) => void;
}

function StatusChip({ isActive = false, activatedText, handleClick }: StatusChipProps) {
  return (
    <Box display="flex" alignItems="center">
      {activatedText && (
        <Chip
          sx={{
            borderRadius: (theme) => theme.shape.customBorderRadius.small,
            backgroundColor: (theme) =>
              isActive ? theme.palette.primary.light : theme.palette.grey[300],
            color: (theme) => (isActive ? theme.palette.primary.darker : theme.palette.grey[900])
          }}
          label={activatedText}
        />
      )}
      <IconButton onClick={handleClick} sx={{ boxShadow: 'none' }}>
        <Iconify
          sx={{ color: (theme: Theme) => theme.palette.grey[900] }}
          icon={'pepicons-pop:dots-y'}
        />
      </IconButton>
    </Box>
  );
}

export default memo(StatusChip);
