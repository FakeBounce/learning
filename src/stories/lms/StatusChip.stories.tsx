import type { Meta, StoryObj } from '@storybook/react';
import StatusChip from '@src/components/lms/StatusChip';
import { Typography } from '@mui/material';

const meta: Meta = {
  title: 'Example/StatusChip',
  component: StatusChip,
  parameters: {
    layout: 'centered',
    waitForSVG: true
  }
};

export default meta;

type Story = StoryObj<typeof StatusChip>;

export const Active: Story = {
  args: {
    isActive: true,
    activatedText: 'Active',
    handleClick: () => alert('Clicked!')
  }
};

export const Inactive: Story = {
  args: {
    isActive: false,
    activatedText: 'Inactive',
    handleClick: () => alert('Clicked!')
  }
};

export const CustomText: Story = {
  args: {
    isActive: true,
    activatedText: <Typography variant="body1">Custom Text</Typography>,
    handleClick: () => alert('Clicked!')
  }
};
