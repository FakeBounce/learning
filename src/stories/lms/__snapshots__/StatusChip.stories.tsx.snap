// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/StatusChip Active smoke-test 1`] = `
<div class="MuiBox-root css-70qvj9">
  <div class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-p8h00o-MuiChip-root">
    <span class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label">
      Active
    </span>
  </div>
  <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-9wiwav-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
  >
    <svg xmlns="http://www.w3.org/2000/svg"
         xmlns:xlink="http://www.w3.org/1999/xlink"
         aria-hidden="true"
         role="img"
         class="MuiBox-root css-1s0zv5d iconify iconify--pepicons-pop"
         width="1em"
         height="1em"
         viewbox="0 0 20 20"
    >
      <g fill="currentColor">
        <circle cx="10"
                cy="15"
                r="2"
        >
        </circle>
        <circle cx="10"
                cy="10"
                r="2"
        >
        </circle>
        <circle cx="10"
                cy="5"
                r="2"
        >
        </circle>
      </g>
    </svg>
    <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
    </span>
  </button>
</div>
`;

exports[`Example/StatusChip CustomText smoke-test 1`] = `
<div class="MuiBox-root css-70qvj9">
  <div class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-p8h00o-MuiChip-root">
    <span class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label">
      <p class="MuiTypography-root MuiTypography-body1 css-gkbk35-MuiTypography-root">
        Custom Text
      </p>
    </span>
  </div>
  <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-9wiwav-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
  >
    <svg xmlns="http://www.w3.org/2000/svg"
         xmlns:xlink="http://www.w3.org/1999/xlink"
         aria-hidden="true"
         role="img"
         class="MuiBox-root css-1s0zv5d iconify iconify--pepicons-pop"
         width="1em"
         height="1em"
         viewbox="0 0 20 20"
    >
      <g fill="currentColor">
        <circle cx="10"
                cy="15"
                r="2"
        >
        </circle>
        <circle cx="10"
                cy="10"
                r="2"
        >
        </circle>
        <circle cx="10"
                cy="5"
                r="2"
        >
        </circle>
      </g>
    </svg>
    <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
    </span>
  </button>
</div>
`;

exports[`Example/StatusChip Inactive smoke-test 1`] = `
<div class="MuiBox-root css-70qvj9">
  <div class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-1c2kyo3-MuiChip-root">
    <span class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label">
      Inactive
    </span>
  </div>
  <button class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-9wiwav-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
  >
    <svg xmlns="http://www.w3.org/2000/svg"
         xmlns:xlink="http://www.w3.org/1999/xlink"
         aria-hidden="true"
         role="img"
         class="MuiBox-root css-1s0zv5d iconify iconify--pepicons-pop"
         width="1em"
         height="1em"
         viewbox="0 0 20 20"
    >
      <g fill="currentColor">
        <circle cx="10"
                cy="15"
                r="2"
        >
        </circle>
        <circle cx="10"
                cy="10"
                r="2"
        >
        </circle>
        <circle cx="10"
                cy="5"
                r="2"
        >
        </circle>
      </g>
    </svg>
    <span class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root">
    </span>
  </button>
</div>
`;
